### PR TITLE
hc process revision

### PR DESCRIPTION
**Pull request by jbathmann**
Revision of HC process in order to account for non constant density model as derived in the pdf attached.
A benchmark in order to compare a analytical steady state solution of the corresponding constant density model to the numerical results is ready. Here I'd like to check the Neumann boundary conditions for component concentration, since they were not working properly beforehand.
 I am not sure how to add and explain this benchmark.
[HC-Process.pdf](https://github.com/ufz/ogs/files/2345953/HC-Process.pdf)

### CODE REVIEW

**Comment by TomFischer** [comment]
Hi Jasper, congratulation to your first PR!

File: ProcessLib/ComponentTransport/ComponentTransportFEM.h [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 #include <Eigen/Dense>
 #include <vector>
-
+#include <iostream>
```
Comment (TomFischer) [inline ProcessLib/ComponentTransport/ComponentTransportFEM.h]: Is this needed somewhere?

File: ProcessLib/ComponentTransport/ComponentTransportFEM.h [hunk 2]
```diff
@@ -33,13 +33,22 @@ template <typename NodalRowVectorType, typename GlobalDimNodalMatrixType>
 struct IntegrationPointData final
 {
     IntegrationPointData(NodalRowVectorType const& N_,
-                         GlobalDimNodalMatrixType const& dNdx_,
+            GlobalDimNodalMatrixType const& dNdx_,
```
Comment (TomFischer) [inline ProcessLib/ComponentTransport/ComponentTransportFEM.h]: Is it planed to use different shape matrices for the p process and the C process? If it is always the same we could save a lot of memory when the implementation would use only one set of shape matrices.

File: ProcessLib/ComponentTransport/ComponentTransportFEM.h (hunk 1 above)
Comment (jbathmann) [reply ProcessLib/ComponentTransport/ComponentTransportFEM.h]: Nope.
Just removed it. Is it possible to add this change to the pull request?

**Comment by jbathmann** [comment]
Sorry for the mess within the commits. I am still a bit confused in how to properly use git... ;)

File: ProcessLib/ComponentTransport/ComponentTransportFEM.h [hunk 3]
```diff
@@ -275,9 +287,9 @@ class LocalAssemblerData : public ComponentTransportLocalAssemblerInterface
         pos.setElementID(_element.getID());
 
         MaterialLib::Fluid::FluidProperty::ArrayType vars;
-
-        auto const p_nodal_values = Eigen::Map<const NodalVectorType>(
-            &local_x[ShapeFunction::NPOINTS], ShapeFunction::NPOINTS);
+        auto const num_nodes = ShapeFunction::NPOINTS;
```
Comment (endJunction) [inline ProcessLib/ComponentTransport/ComponentTransportFEM.h]: @TomFischer Seems there is no difference (in the optimized assembly) whether to use NPOINTS or num_nodes for the Eigen::Map creation.

**Comment by jbathmann** [comment]
Some of the benchmarks have been updated. Additionally a new density model \rho(C,p) = \rho_ref(1+ A_p * p + A_C * C) is added in order to test the new HC process, where the storage parameter has been replaced by:
S= \frac{\phi}{\rho_ref} * \frac{\partial \rho}{\partial p}
Some observations on the benchmarks are documented in the file attached.
[20180912HCTests.pdf](https://github.com/ufz/ogs/files/2379739/20180912HCTests.pdf)

File: Testing/Temporary/CTestCostData.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+---
```
Comment (endJunction) [inline Testing/Temporary/CTestCostData.txt]: spurious file...

**Comment by endJunction** [comment]
There are new tags in the project files; they must be documented.
The pdf is very nice and should be included in the [benchmarks site](https://www.opengeosys.org/docs/benchmarks/hydro-component/elder/).

I can show you the corresponding procedures to include the changes.

**Comment by TomFischer** [comment]
Very nice summary of changes in benchmark results!

File: Tests/Data/Parabolic/ComponentTransport/elder/elder_pcs_0.pvd [hunk 5]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0"?>
```
Comment (TomFischer) [inline Tests/Data/Parabolic/ComponentTransport/elder/elder_pcs_0.pvd]: Did you add this file for comparison reasons?

Comment (jbathmann) [reply Tests/Data/Parabolic/ComponentTransport/elder/elder_pcs_0.pvd]: Nope. Was an "accident"...

**Pull request merged**